website/blog/2020-11/01/_index.md

```diff
@@ -0,0 +1,185 @@
+One reason you stumble upon this blog post could be that you saw errors like the following in your ETCD instances:
+
+```
+etcd-main-0 etcd 2020-09-03 06:00:07.556157 W | etcdserver: read-only range request "key:\"/registry/deployments/shoot--pwhhcd--devcluster2/kube-apiserver\" " with result "range_response_count:1 size:9566" took too long (13.95374909s) to execute
+```
+
+As it turns out, 14 seconds are way too slow for running Kubernetes API servers. It makes them go into the crash loop (leader election fails). Even worse, this whole thing is self-amplifying: The longer a response takes, the more requests queue up, leading to response times increasing further and further. The system is very unlikely to recover. 😞
+
+On Github, you can easily find the reason for this problem. Most probably your disks are too slow (see [etcd-io/etcd#10860](https://github.com/etcd-io/etcd/issues/10860)). So, when you are (like in our case) on GKE and run your ETCD on their default persistent volumes, consider moving from standard disks to SSDs and the error messages should disappear. A guide on how to use SSD volumes on GKE can be found [here](https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/ssd-pd).
+
+Case closed? Well. For some people it might. But when you are seeing this in your Gardener infrastructure, likely, there is something going wrong. The entire ETCD management is fully managed by the Gardener, which makes the problem a bit more interesting to look at. This blog post strives topics such as:
+
+- Gardener operating principles
+- Gardener architecture and ETCD management
+- Pitfalls with multi-cloud environments
+- Migrating GCP volumes to a new storage class
+
+We from metal-stack learned quite a lot about the capabilities of Gardener through this problem. We are happy to share this experience with a broader audience. Gardener adopters and operators read on.
+
+## How Gardener Manages ETCDs
+
+In our infrastructure, we use the Gardener to provision Kubernetes clusters on bare metal machines in our own data centers using <span style="white-space:nowrap">[metal-stack](https://metal-stack.io/)</span>. Even if the entire stack could be running on-premise, our initial [seed cluster](https://gardener.cloud/documentation/concepts/core-components/api-server/#seeds) and the [metal control plane](https://docs.metal-stack.io/stable/overview/architecture/#Metal-Control-Plane) are hosted on GKE. This way, we do not need to manage a single Kubernetes cluster in our entire landscape manually. As soon as we have Gardener deployed on this initial cluster, we can spin up further Seeds in our own data centers through the concept of [shooted seeds](https://github.com/gardener/gardener/blob/master/docs/usage/shooted_seed.md).
+
+To make this easier to understand, let us give you a simplified picture of how our Gardener production setup looks like:
+
+<img title="Production Setup" src="/blog/2020-11/01/01-001.svg"  style="width:80vw; height:auto">
```

Comment:
Is this correct? In the preview, the image is not displayed correctly...
I assume, standard markup image includes don't work here?

Reply:
I stole that from other blog articles of this repository. In my preview this was looking good. :) 

Used: 

```
make serve
``` 

![Screenshot from 2020-11-18 11-01-55](https://user-images.githubusercontent.com/15035165/99515653-85fcd900-298d-11eb-8f0f-b2523040c801.png)

Reply:
Ok, perfect 👍